query/stdlib/influxdata/influxdb/rules_test.go

```diff
@@ -1917,6 +1917,518 @@ func TestPushDownWindowAggregateRule(t *testing.T) {
+					Aggregates:        []plan.ProcedureKind{proc},
+					WindowEvery:       every.Nanoseconds(),
+					CreateEmpty:       createEmpty,
+				}),
+			},
+		}
+		for i, successor := range successors {
+			spec.Nodes = append(spec.Nodes, successor)
+			spec.Edges = append(spec.Edges, [2]int{i, i + 1})
+		}
+		return spec
+	}
+
+	duplicateSpec := func(column, as string) *universe.SchemaMutationProcedureSpec {
+		return &universe.SchemaMutationProcedureSpec{
+			Mutations: []universe.SchemaMutation{
+				&universe.DuplicateOpSpec{
+					Column: execute.DefaultStopColLabel,
+					As:     execute.DefaultTimeColLabel,
+				},
+			},
+		}
+	}
+
+	// ReadRange -> group -> window -> min => ReadWindowAggregate -> group -> min
+	tests = append(tests, plantest.RuleTestCase{
+		Context: haveCaps,
+		Name:    "SimplePassMin",
+		Rules:   rules,
+		Before:  simplePlan(window1m, "min", minProcedureSpec()),
+		After: simpleResult("min", dur1m, false,
+			plan.CreatePhysicalNode("group", groupResult()),
+			plan.CreatePhysicalNode("min", minProcedureSpec()),
+		),
+	})
+
+	// ReadRange -> group -> window -> max => ReadWindowAggregate -> group -> max
+	tests = append(tests, plantest.RuleTestCase{
+		Context: haveCaps,
+		Name:    "SimplePassMax",
+		Rules:   rules,
+		Before:  simplePlan(window1m, "max", maxProcedureSpec()),
+		After: simpleResult("max", dur1m, false,
+			plan.CreatePhysicalNode("group", groupResult()),
+			plan.CreatePhysicalNode("max", maxProcedureSpec()),
+		),
+	})
+
+	// ReadRange -> group -> window -> mean => ReadGroup -> mean
+	// TODO(jsternberg): When we begin pushing down mean calls,
+	// this test will need to be updated to the appropriate pattern.
+	// The reason why this is included is because we cannot rewrite
+	// a grouped mean to use read window aggregate with mean. We
+	// will need this plan to be something different that doesn't
+	// exist yet so this is testing that we don't attempt to use
+	// this planner rule for mean.
+	tests = append(tests, plantest.RuleTestCase{
+		Context: haveCaps,
+		Name:    "SimplePassMean",
+		Rules:   rules,
+		Before:  simplePlan(window1m, "mean", meanProcedureSpec()),
+		After: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+					ReadRangePhysSpec: readRange,
+					GroupMode:         flux.GroupModeBy,
+				}),
+				plan.CreatePhysicalNode("window", &window1m),
+				plan.CreatePhysicalNode("mean", meanProcedureSpec()),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+			},
+		},
+	})
+
+	// ReadRange -> group -> window -> count => ReadWindowAggregate -> group -> sum
+	tests = append(tests, plantest.RuleTestCase{
+		Context: haveCaps,
+		Name:    "SimplePassCount",
+		Rules:   rules,
+		Before:  simplePlan(window1m, "count", countProcedureSpec()),
+		After: simpleResult("count", dur1m, false,
+			plan.CreatePhysicalNode("group", groupResult()),
+			plan.CreatePhysicalNode("count", sumProcedureSpec()),
```

Comment:
I feel this can look confusing. The node is named `count` but has a `sum` spec. I recommend we also rename the node.

Reply:
I'll try to. I couldn't figure out how to do it without using `plan.ReplaceNode` and I wasn't having a lot of success with that specific function.

Reply:
I think I did this successfully.